### PR TITLE
GKE Deployment + Deployment Automation

### PR DESCRIPTION
**Pull request by Hgjnnf**
Automated deployment to Google Cloud.

On push to main branch, this workflow will be triggered.

All the authentication and secrets have been added to this repository.

To test the workflow, you can create a fork repo and push this branch to its main. The workflow should take a bit over two minutes. 

Successful Workflow Completion
image

Deployed Changes After Runnning the Workflow
image

### CODE REVIEW

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      
+    - name: Create .config folder
+      run: mkdir .config
+      
+    - name: create-firebase-secret-json
+      id: create-json-1
+      uses: jsdaniell/create-json@v1.2.2
+      with:
+        name: "firebase_secret_key.json"
+        json: ${{ secrets.FIREBASE_AUTH }}
```
Comment (IcedTea2K) [inline .github/workflows/deploy.yaml:31]: Just curious, is secrets a special variable? Or is it like an environment path?

Comment (Hgjnnf) [reply .github/workflows/deploy.yaml:31]: It's a secret variable. I added them onto GitHub. Should be in the settings

Comment (IcedTea2K) [reply .github/workflows/deploy.yaml:31]: Ah okay! Do you mind uploading some pictures of you testing the changes? Otherwise, everything looks good to me

**Review by Kevin-Tan25 — APPROVED** [review]
LGTM

**Pull request merged**